src/common.js

```diff
@@ -1,29 +1,25 @@
 /* global Nimiq */
 /* global RpcServer */
 
-/**
- * @returns {string}
- */
-function allowedOrigin() {
-    switch (window.location.origin) {
-    case 'https://keyguard-next.nimiq.com': return 'https://accounts.nimiq.com';
-    case 'https://keyguard-next.nimiq-testnet.com': return 'https://accounts.nimiq-testnet.com';
-    default: return '*';
-    }
-}
+/** @type {Promise<void>?} */
+let __nimiqLoaded = null;
 
 /**
+ * Singleton promise
+ *
  * @returns {Promise<void>}
  */
 async function loadNimiq() {
-    try {
+    // eslint-disable-next-line no-return-assign
+    return __nimiqLoaded || (__nimiqLoaded = new Promise(async resolve => {
         // Load web assembly encryption library into browser (if supported)
         await Nimiq.WasmHelper.doImportBrowser();
+
         // Configure to use test net for now
         Nimiq.GenesisConfig.test();
-    } catch (e) {
```

Comment:
In what case does `Nimiq.WasmHelper.doImportBrowser()` throw? Can we handle it, do we need to handle it?

Reply:
`doImportBrowser()` throws when it's already been called before. Which should not happen with this singleton promise:

From https://github.com/nimiq-network/core/blob/master/src/main/generic/utils/WasmHelper.js#L5
```javascript
        if (WasmHelper._importStarted) {
            Log.e(WasmHelper, 'doImportBrowser invoked twice');
            return;
        }
```